src/vs/workbench/services/editor/browser/editorService.ts

```diff
@@ -756,6 +756,17 @@ export class EditorService extends Disposable implements EditorServiceImpl {
 
 	//#endregion
 
+	//#region findEditor()
+
+	getEditorsForResource(resource: URI, group: IEditorGroup): IEditorInput[] {
+		const cononicalResource = this.asCanonicalEditorResource(resource);
+		return group.getEditors(EditorsOrder.SEQUENTIAL).filter(editor => {
```

Comment:
Instead of looping over all editors in the group, I would do a quick first `this.isOpen({ resource })` to see if the editor is open after all. This is a very fast `Set` lookup without loops and can be used to return quickly.